### ISSUE TITLE
Include POI names in routing URL

### ISSUE THREAD

**Issue by Ekleog** [openstreetmap/openstreetmap-website#1910]
Hello,

When I search for a Paris → Lyon path, I come upon this page: https://www.openstreetmap.org/directions?engine=graphhopper_car&route=48.857%2C2.351%3B45.758%2C4.832 (or similar with OSRM, hence my supposition it's an issue with the OSM website, but if you think that's an upstream issue, I can take it to OSRM and/or GraphHopper :))

Unfortunately, if you open it, you won't be able to know that I searched for “Paris” and “Lyon”, and you will only have coordinates available.

I recently wanted to show off OSM to friends by sending them a path to a POI I had recently added that isn't in google map, but the absence of this feature makes it harder for me to do this “elegantly”.

Anyway, thank you for the work you make for OSM, that's a wonderful project!

**Comment by tomhughes** [comment]
I thought this was a duplicate but I can't find it right now.

It's kind of deliberate though, to stop the URLs getting too long. We do try to convert back to a POI name from the coordinates but it doesn't always convert back to the original name.

**Comment by Ekleog** [comment]
Hmm… I can confirm that the converting-back-to-POI-name works on project-osrm.org (eg. [there](http://map.project-osrm.org/?z=10&center=45.645368%2C4.747982&loc=48.856610%2C2.351499&loc=45.757814%2C4.832011&hl=en&alt=0)), but it doesn't work on the main openstreetmap.org website. It would already be half-solving this issue, indeed :)

Also, maybe having an option to have a longer link with the full information could help? Actually, honestly I personally don't care much about the length of addresses in most situations (ie. except when pasting URLs on IRC, where they could be broken). This is thus from my point of view something that is not at feature-parity with google maps (which turns out to have… quite lengthy URLs).

That said, having a choice would likely be best, indeed :) but in most cases when I share a URL I expect people to have the same information as I do, even though I know this is less and less true in today's web.

**Comment by tomhughes** [comment]
We explicitly do not aim for feature parity with google maps - we are not trying to build a full service end user mapping portal.

The web site exists to support mappers rather than end users and routing is primarily there as a tool for detecting broken connections rather than as a way to publish routes to the world.

**Comment by Ekleog** [comment]
Hmm… To me, it sounds like a debatable opinion from a marketing point of view (as the openstreetmap.org website is the first point of access for someone willing to try OSM), but if the decision is conscious, then so be it :)

Leaving this open just because the conversion back to POI names from coordinates appears not to work on openstreetmap.org at the moment.

**Comment by ADepic** [comment]
> The web site exists to support mappers rather than end users and routing is primarily there as a tool for detecting broken connections rather than as a way to publish routes to the world.

I disagree. The main page has to be for consumers otherwise many people will disregard the project as a whole simply from the main page. 

The would be more harmful then helpful to the mapping community due to the fact that many potential users, who woul d have gotten interested and started mapping would be lost. And that means potential usability features will be scattered over many website for end users to scavenge and find, which I doubt anyone will do unless someone experienced tells them.

Not to mention a separate website could be made to "support mappers rather than end users".

**Comment by woodpeck** [comment]
The main page is for mappers, not consumers, because hoping to achieve feature parity with e.g. google maps would simply bind too many resources. The main page is one of many ways to consume OSM data - today, many more people use (or see) our data through third party web sites or mobile apps. If our main web site is a showcase then it is a showcase for the data we have, not a show case for how great the UI designers are that we employ.

**Comment by ADepic** [comment]
I agree osm shouldn't be selling hotels deals and prices like google maps does (AKA. osm shouldn't have to match the feature parity of google maps, but atleast make it user friendly! For example, POIs are not clickable, and even if they were, descriptions of POI's only show change set comments and tags - not user friendly. Changing that does not require expensive employment.

The problem with third party apps and websites is that they do not engage with the project as much, so many users of a mapplication e.g. Maps.me won't find out or care about osm. Yet if you search for osm you find a non-user friendly website. Why does it just have to be showcase when it could be so much more, especially considering it would be so much more useable for consumers with a few small changes...

-------- Original Message --------

**Comment by RDX-Awais** [comment]
f
